third_party/terraform/resources/resource_compute_instance_from_template.go

```diff
@@ -37,6 +37,17 @@ func computeInstanceFromTemplateSchema() map[string]*schema.Schema {
 		s[field].Optional = true
 	}
 
+	// schema.SchemaConfigModeAttr allows these fields to be removed in Terraform 0.12.
+	// Passing field_name = [] in this mode differentiates between an intentionally empty
+	// block vs an ignored computed block.
+	nic := s["network_interface"].Elem.(*schema.Resource)
```

Comment:
do we need to check if any of these are nil before we access/cast?

Reply:
Since they are part of the statically defined Schema I don't think so. And if they are missing I think it's acceptable to panic (which will happen during tests).